helm/examples/azure/log-analytics/loganalytics-hpcc-logaccessV2.yaml

```diff
@@ -0,0 +1,94 @@
+# Configures HPCC logAccess to target Azure Log Analytics Workspace
+global:
+  logAccess:
+    name: "Azure LogAnalytics LogAccess"
+    type: "AzureLogAnalyticsCurl"
+    #connection:
+      #All connection attributes are optional.
+      #It is preferable to provide connection values as secret values category 'esp', secret name 'azure_logaccess'
+      # NOTE: secret 'azure_logaccess' must include 'aad-client-secret' and it cannot be provided in configuration
+      #
+      #workspaceID: "XYZ"      #ID of the Azure LogAnalytics workspace to query logs from
+                               # Secret value equivalent: 'ala-workspace-id'
+      #clientID: "DEF"         #ID of Azure Active Directory registered application with api.loganalytics.io access - format: 00000000-0000-0000-0000-000000000000
+                               # Secret value equivalent: 'aad-client-id'
+      #tenantID: "ABC"         #The Azure Active Directory Tenant ID, required for KQL API access
+                               # Secret value equivalent: 'aad-tenant-id'
+    logMaps:
+    - type: "global"
+      storeName: "ContainerLogV2"
+      searchColumn: "LogMessage"
+      timeStampColumn: "hpcc_log_timestamp"
+      columnType: "dynamic"
+      columnMode: "ALL"
+    - type: "workunits"
+      storeName: "ContainerLogV2"
+      searchColumn: "hpcc_log_jobid"
+      columnMode: "DEFAULT"
+      columnType: "string"
+    - type: "components"
+      storeName: "ContainerLogV2"
+      searchColumn: "ContainerName"
+      columnMode: "DEFAULT"
+      columnType: "string"
+      disableJoins: true
+      projectName: "hpcc_log_component"
+    - type: "audience"
+      searchColumn: "hpcc_log_audience"
+      enumValues:
+      - code: OPR
+      - code: USR
+      - code: PRO
+      - code: ADT
+      columnMode: "DEFAULT"
+      columnType: "enum"
+    - type: "class"
+      searchColumn: "hpcc_log_class"
+      enumValues:
+      - code: DIS
+      - code: ERR
+      - code: WRN
+      - code: INF
+      - code: PRO
+      - code: MET
+      columnMode: "DEFAULT"
+      columnType: "enum"
+    - type: "instance"
```

Comment:
@jakesmith I didn't meant to push this particular change L56-L61, but since it's here we should discuss.
We have the option to filter/sort by podName by virtue of the "instance" category as this change illustrates, or we could add a new dedicated POD category (significantly more complex). Let me know if you'd like to discuss in person.

Reply:
Thought about this a bit more, and related to my comments above under "components".

I think we need 3 entities here:

1) pod (name) - unique, can container multiple instances of same or different types. e.g. "myeclccserver-598466f98c-2w7s8", "dfuserver-68b9976d84-b9l4r" ,"thormanager-job-w20231005-095934-graph1-2wzb4", ...

2) instance (name) - this is the smallest individual entity we track and corresponds to the process with a container, where there is usually only 1 (but not always). These names are usually not globally unique (and can't be, but are if templated jobs), but will be unique per pod. e.g. "eclccserver1", "hthor10", "myeclwatch", "thormanager-job-w20231005-095934-graph1", ...

3) component (type) - the type of the instance. I think we already have this semantically in MSGFIELD_component, but we don't log it. e.g. "eclccserver","hthor", "thorworker", "wu-archiver", "spray-servicee", "eclservices", ...

- pod name is the smallest entity of work in k8s and we are [now] tracking these per job, and they are a fast entry point into logging relating to a single job.
- instance name will be useful when wanting to search for all logging for a particular component instance, e.g. all runs of "hthor10" (across all replicas)
- component type will be useful when wanting to search for all logging for a type of task, e.g. all "hthor" work (across all instances of type "hthor", since there may be multiple in the environment).